File: src/gpt_generator/modules/file_description_architector.py
import os
from gpt_generator.helpers.gpt_requests import create_chat_completion
from helpers.architecture_extractor import extract_file_description, strurcture_to_list_of_files
from helpers.config import config


from models.gpt_responses_interface import ArchitectorResponse


def gpt_file_description_architector(project: ArchitectorResponse, prompt: str) -> list[ArchitectorResponse | str]:

  try:
    project_structure = project.structure
    architector_config = next(
        (architector for architector
         in config.ai_config.architectors if architector.name == "file_architector"),
        None)

    if not architector_config:
      raise ValueError(
          "Architector 'file_architector' not found in config.")

    response: str = create_chat_completion(
        architector_config.model,
        float(architector_config.temperature),
        architector_config.system_prompts + [],
        architector_config.assistant_prompts + [],
        architector_config.user_prompts +
        [prompt, project_structure, str(
            strurcture_to_list_of_files(project_structure))]
    )

    project._files = extract_file_description(response)

    return [project, response]
  except Exception as e:
    print(f"An error occurred: {e} in {__name__}")
    return []

File: src/gpt_generator/modules/file_description_architector.ts
import createChatCompletion from '../helpers/gpt_requests';
import { extractFileDescription, structureToListOfFiles } from '../../helpers/architecture_extractor';
import { config } from '../../helpers/config';
import { ArchitectorResponse } from '../../models/gpt_responses_interface';

const gptFileDescriptionArchitector = async (project: ArchitectorResponse, prompt: string): Promise<[ArchitectorResponse, string]> => {
  try {
    const { structure } = project;
    const architectorConfig = config.ai_config.architectors.find((architector) => architector.name === 'file_architector');

    if (!architectorConfig) {
      throw new Error("Architector 'file_architector' not found in config.");
    }

    const response = await createChatCompletion(
      architectorConfig.model,
      Number(architectorConfig.temperature),
      architectorConfig.system_prompts,
      architectorConfig.assistant_prompts,
      architectorConfig.user_prompts,
      [
        { role: 'user', content: structure },
        { role: 'user', content: JSON.stringify(structureToListOfFiles(structure)) },
        { role: 'user', content: prompt },
      ]
    );

    project.files = extractFileDescription(response);

    return [project, response];
  } catch (e) {
    console.error(`An error occurred: ${e} in ${__filename}`);
    return [] as unknown as [ArchitectorResponse, string];
  }
};

export default gptFileDescriptionArchitector;
